lib/maths/analytics/CBoostedTreeFactory.cc

```diff
@@ -139,11 +172,11 @@ CBoostedTreeFactory::buildForTrain(core::CDataFrame& frame, std::size_t dependen
 
     this->prepareDataFrameForTrain(frame);
 
-    m_TreeImpl->m_InitializationStage != CBoostedTreeImpl::E_NotInitialized
+    m_TreeImpl->m_InitializationStage != CBoostedTreeImpl::E_EncodingInitialized
         ? this->skipProgressMonitoringFeatureSelection()
         : this->startProgressMonitoringFeatureSelection();
 
-    skipIfAfter(CBoostedTreeImpl::E_NotInitialized,
+    skipIfAfter(CBoostedTreeImpl::E_EncodingInitialized,
```

Comment:
The checkpointing logic doesn't seem quite correct. You need to create a checkpoint for encoding by persisting `skipCheckpointIfAtOrAfter`. Also you should never be checking earlier checkpoints after later ones as in the following code  to `selectFeaturesAndEncodeCategories`, etc. Let's discuss this offline.